eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxo.hs

```diff
@@ -289,31 +278,75 @@ instance
   ) =>
   EncCBOR (ConwayUtxoPredFailure era)
   where
-  encCBOR =
-    encode . \case
-      UtxosFailure a -> Sum (UtxosFailure @era) 0 !> To a
-      BadInputsUTxO ins -> Sum (BadInputsUTxO @era) 1 !> To ins
-      OutsideValidityIntervalUTxO a b -> Sum OutsideValidityIntervalUTxO 2 !> To a !> To b
-      MaxTxSizeUTxO a b -> Sum MaxTxSizeUTxO 3 !> To a !> To b
-      InputSetEmptyUTxO -> Sum InputSetEmptyUTxO 4
-      FeeTooSmallUTxO a b -> Sum FeeTooSmallUTxO 5 !> To a !> To b
-      ValueNotConservedUTxO a b -> Sum (ValueNotConservedUTxO @era) 6 !> To a !> To b
-      WrongNetwork right wrongs -> Sum (WrongNetwork @era) 7 !> To right !> To wrongs
-      WrongNetworkWithdrawal right wrongs -> Sum (WrongNetworkWithdrawal @era) 8 !> To right !> To wrongs
-      OutputTooSmallUTxO outs -> Sum (OutputTooSmallUTxO @era) 9 !> To outs
-      OutputBootAddrAttrsTooBig outs -> Sum (OutputBootAddrAttrsTooBig @era) 10 !> To outs
-      OutputTooBigUTxO outs -> Sum (OutputTooBigUTxO @era) 11 !> To outs
-      InsufficientCollateral a b -> Sum InsufficientCollateral 12 !> To a !> To b
-      ScriptsNotPaidUTxO a -> Sum ScriptsNotPaidUTxO 13 !> To a
-      ExUnitsTooBigUTxO a b -> Sum ExUnitsTooBigUTxO 14 !> To a !> To b
-      CollateralContainsNonADA a -> Sum CollateralContainsNonADA 15 !> To a
-      WrongNetworkInTxBody a b -> Sum WrongNetworkInTxBody 16 !> To a !> To b
-      OutsideForecast a -> Sum OutsideForecast 17 !> To a
-      TooManyCollateralInputs a b -> Sum TooManyCollateralInputs 18 !> To a !> To b
-      NoCollateralInputs -> Sum NoCollateralInputs 19
-      IncorrectTotalCollateralField c1 c2 -> Sum IncorrectTotalCollateralField 20 !> To c1 !> To c2
-      BabbageOutputTooSmallUTxO x -> Sum BabbageOutputTooSmallUTxO 21 !> To x
-      BabbageNonDisjointRefInputs x -> Sum BabbageNonDisjointRefInputs 22 !> To x
+  encCBOR pf = ifEncodingVersionAtLeast (natVersion @10) (tenOnwards pf) (beforeTen pf)
+    where
+      tenOnwards =
+        encode . \case
+          UtxosFailure a -> Sum (UtxosFailure @era) 0 !> To a
+          BadInputsUTxO ins -> Sum (BadInputsUTxO @era) 1 !> To ins
+          OutsideValidityIntervalUTxO a b -> Sum OutsideValidityIntervalUTxO 2 !> To a !> To b
+          MaxTxSizeUTxO mm -> Sum MaxTxSizeUTxO 3 !> To mm
+          InputSetEmptyUTxO -> Sum InputSetEmptyUTxO 4
+          FeeTooSmallUTxO mm -> Sum FeeTooSmallUTxO 5 !> To mm
+          ValueNotConservedUTxO mm -> Sum (ValueNotConservedUTxO @era) 6 !> To mm
+          WrongNetwork right wrongs -> Sum (WrongNetwork @era) 7 !> To right !> To wrongs
+          WrongNetworkWithdrawal right wrongs -> Sum (WrongNetworkWithdrawal @era) 8 !> To right !> To wrongs
```

Comment:
You mean for `WrongNetworkWithdrawal`? I think it is the correct order, since the first argument to the construcor is the correct network ID and the second one is a set of wrong network IDs present in the transaction body. Since the types are different we cannot use the `Mismatch` type here and we also don't need to.